src/Underscores.jl

```diff
@@ -68,7 +68,12 @@ replace__(ex) = add_closures(ex, "__", r"^__([0-9]*|[₀-₉]*)$")
 # by storing a per-module _pipeline_ops in the module using @_.)
 const _pipeline_ops = [:|>, :<|, :∘]
 
-function lower_underscores(ex)
+const _square_bracket_ops = [:comprehension, :typed_comprehension, :generator,
+                             :vcat, :typed_vcat, :hcat, :typed_hcat, :row]
+
+_isoperator(x) = x isa Symbol && Base.isoperator(x)
+
+function lower_underscores(ex, replace__=replace__)
```

Comment:
This `replace__=replace__` seems a bit ugly. Perhaps the function should be split in two, one of which catches pipelining & otherwise applies `replace__`, the other deals with where to apply `replace_`.

Reply:
I made a new `lower_innner()` function to simplify this now.